frontend/src/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionRegisteredFromLink.tsx

```diff
@@ -0,0 +1,174 @@
+}) => {
+  const { projects } = React.useContext(ProjectsContext);
+
+  // Extract information using utility functions
+  const registeredFromCatalogDetails = modelSourcePropertiesToCatalogParams(
+    modelArtifact.modelSourceKind,
+    modelArtifact.modelSourceClass,
+    modelArtifact.modelSourceGroup,
+    modelArtifact.modelSourceName,
+    modelArtifact.modelSourceId,
+  );
+
+  const registeredFromPipelineDetails = modelSourcePropertiesToPipelineRunRef(
+    modelArtifact.modelSourceKind,
+    modelArtifact.modelSourceGroup,
+    modelArtifact.modelSourceId,
+    modelArtifact.modelSourceName,
+  );
+
+  // If no source information is available, don't render anything
+  if (!registeredFromCatalogDetails && !registeredFromPipelineDetails) {
+    return null;
+  }
+
+  // Generate the catalog URL if we have catalog details
+  const catalogModelUrl = registeredFromCatalogDetails
+    ? getCatalogModelDetailsRoute(registeredFromCatalogDetails)
+    : '';
+
+  // For pipeline source, look up the project
+  const registeredPipelineProject =
+    registeredFromPipelineDetails &&
+    projects.find((p) => p.metadata.name === registeredFromPipelineDetails.project);
+
+  // If we have pipeline details, render the pipeline link
+  if (registeredFromPipelineDetails) {
+    const { project, runId, runName } = registeredFromPipelineDetails;
+
+    // If we also have a catalog URL, link to the catalog
+    const renderRunLink = catalogModelUrl ? (
+      <Link
+        to={catalogModelUrl}
+        style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}
+        data-testid="catalog-model-link"
+      >
+        {runName}
+      </Link>
+    ) : registeredPipelineProject ? (
+      <Link
+        style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}
+        to={globalPipelineRunDetailsRoute(project, runId)}
+      >
+        {runName}
+      </Link>
+    ) : (
+      <b>{runName}</b>
+    );
+
+    const renderProject = registeredPipelineProject
+      ? registeredPipelineProject.metadata.annotations?.['openshift.io/display-name'] ??
+        registeredPipelineProject.metadata.name
+      : project;
+
+    return (
+      <Flex
+        spaceItems={{ default: 'spaceItemsXs' }}
+        alignItems={{ default: 'alignItemsCenter' }}
+        data-testid="registered-from-pipeline"
+      >
+        <FlexItem data-testid="pipeline-run-link">Run {renderRunLink} in</FlexItem>
+        <FlexItem style={{ display: 'flex' }}>
+          <TypedObjectIcon
+            resourceType={ProjectObjectType.project}
+            style={{ height: 24, width: 24 }}
+          />
+        </FlexItem>
+        <FlexItem>
+          <b>{renderProject}</b>
+        </FlexItem>
+        {catalogModelUrl && (
+          <FlexItem data-testid="catalog-model-info">
+            <span style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}>
+              (from Model Catalog)
+            </span>
+          </FlexItem>
+        )}
```

Comment:
Same here - it doesn't make sense to render details about both pipeline and catalog, a model will be registered from either one or the other.